web/src/main/java/it/pagopa/selfcare/pagopa/backoffice/web/controller/UtilsController.java

```diff
@@ -26,31 +37,93 @@ public class UtilsController {
+        PaymentServiceProviderDetailsResource paymentServiceProviderDetailsResource = null;
+        BrokerPspDetails brokerPspDetails;
+        PaymentServiceProviderDetails paymentServiceProviderDetails;
+        try {
+            brokerPspDetails = apiConfigService.getBrokerPsp(brokerPspCode);
+            brokerPspDetailsResource = ChannelMapper.toResource(brokerPspDetails);
+        } catch (Exception e) {
+            log.trace("getBrokerAndPspDetails - Not BrokerPSP found");
+        }
 
-        PaymentServiceProviderDetails paymentServiceProviderDetails = apiConfigService.getPSPDetails(brokerPspCode);
-        PaymentServiceProviderDetailsResource paymentServiceProviderDetailsResource = ChannelMapper.toResource(paymentServiceProviderDetails);
+        try {
+            paymentServiceProviderDetails = apiConfigService.getPSPDetails(brokerPspCode);
+            paymentServiceProviderDetailsResource = ChannelMapper.toResource(paymentServiceProviderDetails);
+        } catch (Exception e) {
+            log.trace("getBrokerAndPspDetails - Not PaymentServiceProvider found");
+        }
+
+        if (brokerPspDetailsResource == null && paymentServiceProviderDetailsResource == null) {
+            throw new ResourceNotFoundException("Nessun dato trovato per il broker o per il creditorInstitution");
+        }
 
         BrokerOrPspDetailsResource resource = new BrokerOrPspDetailsResource();
         resource.setBrokerPspDetailsResource(brokerPspDetailsResource);
         resource.setPaymentServiceProviderDetailsResource(paymentServiceProviderDetailsResource);
 
-        
-        
+        log.debug(LogUtils.CONFIDENTIAL_MARKER, "getBrokerAndPspDetails result = {}", resource);
+        log.trace("getBrokerAndPspDetails end");
+
+        return resource;
+    }
+
+    @GetMapping(value = "/ec-brokers/{code}/details", produces = {MediaType.APPLICATION_JSON_VALUE})
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "", notes = "${swagger.api.channels.getBrokerPsp}")
+    public BrokerAndEcDetailsResource getBrokerAndEcDetails(@ApiParam("swagger.request.brokerEcCode")
+                                                            @PathVariable(required = true, name = "code") String brokerEcCode) throws Exception {
+        log.trace("getBrokerOrEcDetails start");
+        String xRequestId = UUID.randomUUID().toString();
+        log.debug("getBrokerOrEcDetails brokerPspCode = {} , xRequestId:  {}", brokerEcCode, xRequestId);
+
+        BrokersResource brokersResource = new BrokersResource();
+        brokersResource.setBrokers(new ArrayList<>());
+        CreditorInstitutionDetailsResource creditorInstitutionDetailsResource = null;
+        Brokers brokers;
+        CreditorInstitutionDetails creditorInstitutionDetails;
+
+        try {
+            brokers = apiConfigService.getBrokersEC(1, 0, brokerEcCode, null, null, "ASC");
+            brokersResource = BrokerMapper.toResource(brokers);
+        } catch (Exception e) {
```

Comment:
use a specific exception (for example `FeignException.NotFound`)

Reply:
fixed